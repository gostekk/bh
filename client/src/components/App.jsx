import React from 'react';
import PropTypes from 'prop-types';

// Material-ui
import CssBaseline from '@material-ui/core/CssBaseline';
import Paper from '@material-ui/core/Paper';
import withStyles from '@material-ui/core/styles/withStyles';

import Form from '../containers/Form';

const propTypes = {
  classes: PropTypes.shape({
    layout: PropTypes.string.isRequired,
    paper: PropTypes.string.isRequired,
  }).isRequired,
};

const styles = theme => ({
  layout: {
    width: 'auto',
    minWidth: 400,
    display: 'block', // Fix IE11 issue.
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`,
  },
});

export const App = ({ classes }) => (
  <React.Fragment>
    <CssBaseline />
    <main className={classes.layout}>
      <Paper className={classes.paper}>
        <Form />
      </Paper>
    </main>
  </React.Fragment>
);

App.propTypes = propTypes;

export default withStyles(styles)(App);
