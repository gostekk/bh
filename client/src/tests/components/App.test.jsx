import React from 'react';
import { shallow } from 'enzyme';
import App from '../../components/App.jsx';

describe('App Component', () => {
  it('should render App component and match to snapshot', () => {
    const wrapper = shallow(<App />).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
