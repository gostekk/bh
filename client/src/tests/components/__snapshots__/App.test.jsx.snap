// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render App component and match to snapshot 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <main
    className="Component-layout-1"
  >
    <WithStyles(Paper)
      className="Component-paper-2"
    >
      <Connect(WithStyles(Form)) />
    </WithStyles(Paper)>
  </main>
</Fragment>
`;
